### PR TITLE
lp-106 : add error handler

### PR DESCRIPTION
**Pull request by lduranteau**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-106

### Change description
add error handler - 404 and 500

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by markpit** [comment]
what is the preference for the file names? I used start.controller.ts but see you have eg. error-handler.ts. Should we change start.controller.ts to start-controller.ts?

File: src/middleware/error-handler.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    templateName: config.NOT_FOUND_TEMPLATE
+  });
+};
+
+/**
+ * This handler catches any other error thrown within the application.
+ * Use this error handler by calling next(e) from within a controller
+ * Always keep this as the last handler in the chain for it to work.
+ */
+const errorHandler = (err: Error, req: Request, res: Response, _next: NextFunction) => {
```
Comment (mattch1) [inline src/middleware/error-handler.ts:16]: Are there any jest tests that can be added for this?

**Comment by markpit** [comment]
> what is the preference for the file names? I used start.controller.ts but see you have eg. error-handler.ts. Should we change start.controller.ts to start-controller.ts?

Ignore this comment - The file naming is a wider issue not specifically related to this PR, we'll probably have to have another PR to change them.

File: src/middleware/error-handler.ts (hunk 1 above)
Comment (lduranteau) [reply src/middleware/error-handler.ts]: there is one

**Comment by lduranteau** [comment]
> > what is the preference for the file names? I used start.controller.ts but see you have eg. error-handler.ts. Should we change start.controller.ts to start-controller.ts?
> 
> Ignore this comment - The file naming is a wider issue not specifically related to this PR, we'll probably have to have another PR to change them.

name updated

**Review by mattch1 — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**